types/src/main.ts

```diff
@@ -4087,6 +4087,153 @@ export namespace InlayHint {
 	}
 }
 
+/**
+ * A string value used as a snippet is a template which allows to insert text
+ * and to control the editor cursor when insertion happens.
+ *
+ * A snippet can define tab stops and placeholders with `$1`, `$2`
+ * and `${3:foo}`. `$0` defines the final tab stop, it defaults to
+ * the end of the snippet. Variables are defined with `$name` and
+ * `${name:default value}`.
+ *
+ * @since 3.18.0
+ * @proposed
+ */
+export interface StringValue {
+	/**
+	 * The kind of string value.
+	 */
+	kind: 'snippet';
+	/**
+	 * The snippet string.
+	 */
+	value: string;
+}
+
+export namespace StringValue {
+	export function create(value: string): StringValue {
+		return { value, kind: 'snippet' };
+	}
+}
+
+/**
+ * An inline completion item represents a text snippet that is proposed inline to complete text that is being typed.
+ *
+ * @since 3.18.0
+ * @proposed
+ */
+export interface InlineCompletionItem {
+	/**
+	 * The text to replace the range with. Must be set.
+	 */
+	insertText: string | StringValue;
+
+	/**
+	 * A text that is used to decide if this inline completion should be shown. When `falsy` the {@link InlineCompletionItem.insertText} is used.
+	 */
+	filterText?: string;
+
+	/**
+	 * The range to replace. Must begin and end on the same line.
+	 */
+	range?: Range;
+
+	/**
+	 * An optional {@link Command} that is executed *after* inserting this completion.
+	 */
+	command?: Command;
+}
+
+export namespace InlineCompletionItem {
+	export function create(insertText: string | StringValue, filterText?: string, range?: Range, command?: Command): InlineCompletionItem {
+		return { insertText, filterText, range, command };
+	}
+}
+
+/**
+ * Represents a collection of {@link InlineCompletionItem inline completion items} to be presented in the editor.
+ */
```

Comment:
@since 3.18.0
@proposed 

missing

Reply:
@c-claeys very nice work. Highly appreciated. One last minor change request for a missing @since / @proposed.

Reply:
Thank you and you're welcome. Of course, sorry I missed it. Done.